### PR TITLE
Add Route53 + CloudFormation + SQS + StepFunction + S3 report generation

### LINKED ISSUE #316: Add more resources report (for Tagbot)
Add more report generation of some resources that are often used.

### PR DESCRIPTION
**Pull request by theocerutti**
Add report generation for:

- [X] StepFunctions
- [X] SQS
- [X] S3
- [X] Route 53
- [X] CloudFormation

NOTE:
Tagbot Role Stack required more policy to fetch above resources data.
Then, if you want to test this PR NOW it will not work because current TagBot role are not set.
TagbotRole policies to add during deployment:

```yml
AWSTemplateFormatVersion: 2010-09-09
Description: Create a View Only Access role for tagbot (belong to Trackit/MSolutions.io) and outputs the ARN
Resources:
  TagbotCrossAccountRole:
    Type: 'AWS::IAM::Role'
    Properties:
      RoleName: "TagbotViewAccess"
      AssumeRolePolicyDocument:
        Statement:
          - Action: 'sts:AssumeRole'
            Effect: Allow
            Principal:
              AWS: 'arn:aws:iam::394125495069:root'
      ManagedPolicyArns:
        - 'arn:aws:iam::aws:policy/job-function/ViewOnlyAccess'
        - 'arn:aws:iam::aws:policy/AmazonRDSReadOnlyAccess'
        - 'arn:aws:iam::aws:policy/AmazonESReadOnlyAccess'
      Policies:
        - PolicyName: 'tagbot-view-access'
          PolicyDocument:
            Version: '2012-10-17'
            Statement:
              - Effect: Allow
                Action:
                  - 'sqs:ListQueues'
                  - 'sqs:ListQueueTags'
                  - 'states:ListStateMachines'
                  - 'states:ListTagsForResource'
                  - 's3:ListBucket'
                  - 's3:GetBucketTagging'
                  - 's3:GetBucketLocation'
                  - 'route53:ListHostedZones'
                  - 'route53:ListTagsForResource'
                  - 'cloudformation:ListStacks'
                  - 'cloudformation:DescribeStacks'
                Resource: '*'
Outputs:
  TagbotRoleArn:
    Description: The ARN of the role created for the tagbot web application
    Value: !GetAtt 
      - TagbotCrossAccountRole
      - Arn
```

### CODE REVIEW

File: aws/usageReports/route53/dailyReport.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		logger.Error("Error when getting account id", err.Error())
+		return err
+	}
+	regions, err := utils.FetchRegionsList(ctx, defaultSession)
+	if err != nil {
+		logger.Error("Error when fetching regions list", err.Error())
+		return err
+	}
+	hostedZonesChans := make([]<-chan HostedZone, 0, len(regions))
+	for _, region := range regions {
```
Comment (simonmeyerrr) [inline aws/usageReports/route53/dailyReport.go:89]: It seems to me that route53 works globally and not by region, to be confirmed and corrected if this is the case.

File: aws/usageReports/s3/dailyReport.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		logger.Error("Error when getting account id", err.Error())
+		return err
+	}
+	regions, err := utils.FetchRegionsList(ctx, defaultSession)
+	if err != nil {
+		logger.Error("Error when fetching regions list", err.Error())
+		return err
+	}
+	bucketChans := make([]<-chan Bucket, 0, len(regions))
+	for _, region := range regions {
```
Comment (simonmeyerrr) [inline aws/usageReports/s3/dailyReport.go]: There is probably a way to optimize as ListBucket list buckets from all regions.

File: aws/usageReports/stepfunction/utils.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+import (
+	"context"
+	"crypto/md5"
+	"encoding/base64"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/trackit/jsonlog"
+	taws "github.com/trackit/trackit/aws"
```
Comment (simonmeyerrr) [inline aws/usageReports/stepfunction/utils.go:27]: missing empty line

File: db/migration/0053_add_resources_errors_to_account_update_job.sql [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+--   distributed under the License is distributed on an "AS IS" BASIS,
+--   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+--   See the License for the specific language governing permissions and
+--   limitations under the License.
+
+ALTER TABLE aws_account_update_job ADD stepFunctionError VARCHAR(255) NOT NULL DEFAULT "";
+ALTER TABLE aws_account_update_job ADD s3Error VARCHAR(255) NOT NULL DEFAULT "";
+ALTER TABLE aws_account_update_job ADD sqsError VARCHAR(255) NOT NULL DEFAULT "";
+ALTER TABLE aws_account_update_job ADD cloudFormationError VARCHAR(255) NOT NULL DEFAULT "";
+ALTER TABLE aws_account_update_job ADD route53Error VARCHAR(255) NOT NULL DEFAULT "";
```
Comment (simonmeyerrr) [inline db/migration/0053_add_resources_errors_to_account_update_job.sql]: missing new line

File: db/schema.sql [hunk 5]
```diff
@@ -1207,3 +1207,23 @@ ALTER TABLE tagbot_user ADD stripe_payment_method_identifier VARCHAR(255) NOT NU
...
+--   distributed under the License is distributed on an "AS IS" BASIS,
+--   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+--   See the License for the specific language governing permissions and
+--   limitations under the License.
+
+ALTER TABLE aws_account_update_job ADD stepFunctionError VARCHAR(255) NOT NULL DEFAULT "";
+ALTER TABLE aws_account_update_job ADD s3Error VARCHAR(255) NOT NULL DEFAULT "";
+ALTER TABLE aws_account_update_job ADD sqsError VARCHAR(255) NOT NULL DEFAULT "";
+ALTER TABLE aws_account_update_job ADD cloudFormationError VARCHAR(255) NOT NULL DEFAULT "";
+ALTER TABLE aws_account_update_job ADD route53Error VARCHAR(255) NOT NULL DEFAULT "";
```
Comment (simonmeyerrr) [inline db/schema.sql]: missing new line

File: server/taskProcessAccount.go [hunk 6]
```diff
@@ -23,8 +23,8 @@ import (
 	"time"
 
 	"github.com/trackit/jsonlog"
-
```
Comment (simonmeyerrr) [inline server/taskProcessAccount.go:25]: put again this empty line please

File: tagging/stepfunction/process.go [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	regionForURL := utils.GetRegionForURL(source.Step.Region)
+
+	document := utils.TaggingReportDocument{
+		Account:      source.Account,
+		ResourceID:   source.Step.Name,
+		ResourceType: resourceTypeString,
+		Region:       source.Step.Region,
+		URL:          fmt.Sprintf(urlFormat, regionForURL, regionForURL, source.Account, source.Step.Name),
```
Comment (simonmeyerrr) [inline tagging/stepfunction/process.go]: You have directly ARN in your struct, you don't need to generate it again

**Review by simonmeyerrr — CHANGES_REQUESTED** [review]
We should think about a way to update the role easily for users.
Please update policies in `/policies` folder.

File: aws/usageReports/stepfunction/utils.go (hunk 3 above)
Comment (GabrielRavier) [reply aws/usageReports/stepfunction/utils.go:27]: Should be fixed now

File: db/migration/0053_add_resources_errors_to_account_update_job.sql (hunk 4 above)
Comment (GabrielRavier) [reply db/migration/0053_add_resources_errors_to_account_update_job.sql]: Should be fixed now

File: db/schema.sql (hunk 5 above)
Comment (GabrielRavier) [reply db/schema.sql]: Should be fixed now

File: server/taskProcessAccount.go (hunk 6 above)
Comment (GabrielRavier) [reply server/taskProcessAccount.go:25]: Should be fixed now

File: aws/usageReports/route53/dailyReport.go (hunk 1 above)
Comment (GabrielRavier) [reply aws/usageReports/route53/dailyReport.go:89]: Can confirm that Route53 has at the very least 2 regions, with the second one being in China, as can be seen on this page: https://docs.aws.amazon.com/general/latest/gr/r53.html

(meaning route53 must logically work by region)

File: aws/usageReports/s3/dailyReport.go (hunk 2 above)
Comment (GabrielRavier) [reply aws/usageReports/s3/dailyReport.go]: https://github.com/trackit/trackit/pull/318/commits/de0d7b0608609802d8ff0e60f1d46bb75a640a86 divides the amount of API calls by (approx) the amount of regions

File: tagging/stepfunction/process.go (hunk 7 above)
Comment (GabrielRavier) [reply tagging/stepfunction/process.go]: https://github.com/trackit/trackit/pull/318/commits/45cc4ef6315371132ca8d733e86e89d8414c4831 solves this by using the one from ES

File: aws/usageReports/cloudformation/metricsStats.go [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	res := make([]utils.Tag, 0)
+	tags, err := svc.DescribeStacks(&cloudformation.DescribeStacksInput{
+		StackName: stack.StackName,
+	})
+	if err != nil {
+		logger.Error("Failed to get CloudFormation tags", err.Error())
+		return res
+	}
+
```
Comment (J3suis) [inline aws/usageReports/cloudformation/metricsStats.go:38]: This empty line must be here ?

Comment (EthanHusser) [inline aws/usageReports/cloudformation/metricsStats.go:38]: Maybe just remove this empty line

Comment (GabrielRavier) [reply aws/usageReports/cloudformation/metricsStats.go:38]: It's not mandatory or anything, but it makes the code easier to read IMO

**Review by J3suis — APPROVED** [review]
Everything seems Fine, but there is a lot of line skipping

File: aws/usageReports/cloudformation/metricsStats.go (hunk 8 above)
Comment (GabrielRavier) [reply aws/usageReports/cloudformation/metricsStats.go:38]: Already mentioned by Paul

**Review by EthanHusser — APPROVED** [review]
Nothing to say about the code, just pay attention to some places where there are empty lines.

**Review by simonmeyerrr — APPROVED** [review]

**Pull request merged**